src/DataAccess/DbalSubscriptionRepository.php

```diff
@@ -19,22 +22,37 @@ public function __construct( EntityManager $entityManager ) {
 		$this->entityManager = $entityManager;
 	}
 
+	/**
+	 * @param Subscription $subscription
+	 * @throws SubscriptionRepositoryException
+	 */
 	public function storeSubscription( Subscription $subscription ) {
-		$this->entityManager->persist( $subscription );
-		$this->entityManager->persist( $subscription->getAddress() );
-		$this->entityManager->flush();
+		try {
+			$this->entityManager->persist( $subscription );
+			$this->entityManager->persist( $subscription->getAddress() );
```

Comment:
As far as I can tell, the `persist` methods never throw an `ORMException`. They do throw some thing driving from `InvalidArgumentException`. .... ..... ok, it can throw this, the Doctine docs are just fail. Gonna submit a PR there.